---
title: "Trajectories PGCs- MYC tumors"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r setup_environment, message=FALSE, warning=FALSE}
library(Seurat)
library(monocle3)
library(ggplot2)
library(sctransform)
library(RColorBrewer)
library(readxl)
library(dplyr)
library(reshape2)
library(WriteXLS)
library(Hmisc)

setwd("/marta_home/scRNAseq_Ranalysis/Seurat_v3/trajectories/PGCs")


load("/marta_home/scRNAseq_Ranalysis/Seurat_v3/trajectories/PGCs/workspace.RData")
```


# Load data

```{r cnug}

# PGCs from Marioni
pgc.marioni <- readRDS("/marta_home/scRNAseq_Ranalysis/Seurat_v3/Embryos/published_data/v3/Marioni/full_marioni_integrated_PGCs.rds")

# PGCs from Cao
pgc.cao <- readRDS("/marta_home/scRNAseq_Ranalysis/Seurat_v3/Embryos/published_data/v3/Cao/Cao_PGCs.rds")


# MYC IC
myc.ic <- readRDS("/marta_home/scRNAseq_Ranalysis/Seurat_v3/mouseATRT/v3/ATRT_IC_MYC/myc_ic_tumor_cells.rds")


# MYC EC
myc.ec <- readRDS("/marta_home/scRNAseq_Ranalysis/Seurat_v3/mouseATRT/v3/MRT_EC_MYC/myc_ec_tumor_cells.rds")


# MYC spinal
myc.spinal <- readRDS("/marta_home/scRNAseq_Ranalysis/Seurat_v3/mouseATRT/v3/ATRT_spinal_MYC/myc_spinal_tumor_cells.rds")


# adding labels for trajectory
pgc.cao$stage <- paste0("E", pgc.cao$day)
myc.ic$stage <- "MYC-IC"
myc.ec$stage <- "MYC-EC"
myc.spinal$stage <- "MYC-spinal"

pgc.marioni$batch <- "marioni"
pgc.cao$batch <- "cao"
myc.ic$batch <- "MYC-IC"
myc.ec$batch <- "MYC-EC"
myc.spinal$batch <- "MYC-spinal"


# adding differential analysis labels (per time point)
pgc.marioni$de_id <- pgc.marioni$stage
pgc.cao$de_id <- pgc.cao$day
myc.ic$de_id <- "tumor"
myc.ec$de_id <- "tumor"
myc.spinal$de_id <- "tumor"

# adding differential analysis labels (tumor vs normal)
pgc.marioni$orig.ident <- "marioni"
myc.ic$orig.ident <- "MYC_IC"
myc.ec$orig.ident <- "MYC_EC"
myc.spinal$orig.ident <- "MYC_spin"

myc.list <- list(MYC_IC = myc.ic, MYC_EC = myc.ec, MYC_SPINAL = myc.spinal)
```

## Saving objects for Scanpy

```{r nfo}
pgc.marioni.py <- as.SingleCellExperiment(pgc.marioni, assay = "RNA")
saveRDS(pgc.marioni.py, file = "pgc_marioni_py.rds")

pgc.cao.py <- as.SingleCellExperiment(pgc.cao, assay = "RNA")
saveRDS(pgc.cao.py, file = "pgc_cao_py.rds")

myc.ic.py <- as.SingleCellExperiment(myc.ic, assay = "RNA")
saveRDS(myc.ic.py, file = "MYC_IC_py.rds")

myc.ec.py <- as.SingleCellExperiment(myc.ec, assay = "RNA")
saveRDS(myc.ec.py, file = "MYC_EC_py.rds")

myc.spin.py <- as.SingleCellExperiment(myc.spinal, assay = "RNA")
saveRDS(myc.spin.py, file = "MYC_spin_py.rds")
```

## DE analysis tumors VS normal cells

```{r cnifoga}
pgcs <- merge(pgc.marioni, pgc.cao)

for(j in 1:length(myc.list)){
  tum.obj <- myc.list[[j]]
  merged.obj <- merge(pgcs, tum.obj)
  Idents(merged.obj) <- merged.obj$de_id
  time.points <- setdiff(levels(Idents(merged.obj)), c("tumor", "mixed_gastrulation"))
  for (t in time.points){ 
    markers <- FindMarkers(merged.obj, ident.1 = "tumor", ident.2 = t, test.use = "MAST")
    markers <- markers[markers$p_val_adj < 0.05, ]
    markers <- markers[!is.na(markers$p_val_adj), ]
    WriteXLS(markers, ExcelFileName = paste0(names(myc.list[j]), "_markers_vs_", t, ".xlsx"), row.names = T)
  }
}

tiff("marioni_nGenes_by_stage.tiff", width = 700, height = 600)
VlnPlot(pgc.marioni, features = "nFeature_RNA", group.by = "stage")
dev.off()
```

## DE analysis Marioni

```{r cnduig}
pgc.mar.de <- subset(pgc.marioni, subset = stage != "mixed_gastrulation")
Idents(pgc.mar.de) <- pgc.mar.de$stage

for (i in levels(Idents(pgc.mar.de))){ 
    markers <- FindMarkers(pgc.mar.de, ident.1 = i, test.use = "MAST")
    markers <- markers[markers$p_val_adj < 0.05, ]
    markers <- markers[!is.na(markers$p_val_adj), ]
    WriteXLS(markers, ExcelFileName = paste0("Marioni_markers_", i, ".xlsx"), row.names = T)
  }

# all pgcs marioni vs all tumor cells (by subgroup)
for(j in 1:length(myc.list)){
  tum.obj <- myc.list[[j]]
  merged.obj <- merge(pgc.marioni, tum.obj)
  Idents(merged.obj) <- merged.obj$orig.ident
  markers <- FindMarkers(merged.obj, ident.1 = "marioni", test.use = "MAST")
  markers <- markers[markers$p_val_adj < 0.05, ]
  markers <- markers[!is.na(markers$p_val_adj), ]
  WriteXLS(markers, ExcelFileName = paste0("DE_PCGs_marioni_vs_", names(myc.list[j]), ".xlsx"), row.names = T)
}


### all pgcs marioni vs each tumor cluster
# adding differential analysis labels 
pgc.marioni$clust_de <- "marioni"
myc.ic$clust_de <- myc.ic$new_clustering
myc.ec$clust_de <- myc.ec$new_clustering
myc.spinal$clust_de <- myc.spinal$new_clustering 

myc.list <- list(MYC_IC = myc.ic, MYC_EC = myc.ec, MYC_SPINAL = myc.spinal)

for(j in 1:length(myc.list)){
  tum.obj <- myc.list[[j]]
  merged.obj <- merge(pgc.marioni, tum.obj)
  Idents(merged.obj) <- merged.obj$clust_de
  for(c in unique(tum.obj$clust_de)){
    markers <- FindMarkers(merged.obj, ident.1 = "marioni", ident.2 = c, test.use = "MAST")
    markers <- markers[markers$p_val_adj < 0.05, ]
    markers <- markers[!is.na(markers$p_val_adj), ]
    WriteXLS(markers, ExcelFileName = paste0("DE_PCGs_marioni_vs_", names(myc.list[j]), "_clust",c, ".xlsx"), row.names = T)
  }
  
}

```

## DE analysis Cao

```{r cnduig}
Idents(pgc.cao) <- pgc.cao$day

for (i in levels(Idents(pgc.cao))){ 
    markers <- FindMarkers(pgc.cao, ident.1 = i, test.use = "MAST")
    markers <- markers[markers$p_val_adj < 0.05, ]
    markers <- markers[!is.na(markers$p_val_adj), ]
    WriteXLS(markers, ExcelFileName = paste0("Cao_PGC_markers_", i, ".xlsx"), row.names = T)
  }



```


# Create monocle3 object from Seurat normalized data

```{r dsno}
gene_module_df_list <- list()

for(i in 1:length(myc.list)){
  pgc.mar.data <- GetAssayData(pgc.marioni, slot = "data")
  pgc.cao.data <- GetAssayData(pgc.cao, slot = "data")
  
  subgroup <- names(myc.list[i])
  myc.data <- GetAssayData(myc.list[[i]], slot = "data")
  
  # keep only intersecting genes
  genes.intersecting <- intersect(rownames(pgc.mar.data), rownames(pgc.cao.data))
  genes.intersecting <- intersect(genes.intersecting, rownames(myc.data))

  pgc.mar.data <- pgc.mar.data[genes.intersecting,]
  pgc.cao.data <- pgc.cao.data[genes.intersecting,]
  myc.data <- myc.data[genes.intersecting,]

  merged.data <- cbind(pgc.mar.data, pgc.cao.data)
  merged.data <- cbind(merged.data, myc.data)

  cell.metadata <- data.frame(orig_id = c(rep("PGCs_marioni", ncol(pgc.mar.data)), rep("PGCs_cao",                                           ncol(pgc.cao.data)), rep(subgroup, ncol(myc.data))), 
                              time_point = c(as.character(pgc.marioni$stage), as.character(pgc.cao$day),
                                             rep(NA, ncol(myc.data))), 
                              seu_cluster = c(rep(NA, ncol(pgc.mar.data)), rep(NA, ncol(pgc.cao.data)),
                                  as.character(myc.list[[i]]$new_clustering)), row.names = colnames(merged.data))

  gene.metadata <- data.frame(gene_short_name = rownames(merged.data), row.names = rownames(merged.data))

  # Make the CDS object
  cds <- new_cell_data_set(merged.data, cell_metadata = cell.metadata, gene_metadata = gene.metadata)

  # Preprocessing
  cds <- preprocess_cds(cds, num_dim = 100, norm_method = "none", method = "PCA")
  cds <- align_cds(cds, alignment_group = "orig_id")
                 
  # Dimensionality reduction and trajectories

  cds <- reduce_dimension(cds, reduction_method = "UMAP")
  tiff(paste0(subgroup, "_PGC_umap.tiff"), width = 700, height = 600)
  plot_cells(cds, label_groups_by_cluster=FALSE,  color_cells_by = "orig_id", cell_size = 1.5, 
           group_label_size = 7)
  dev.off()

  cds <- cluster_cells(cds)
  cds <- learn_graph(cds)

  tiff(paste0(subgroup,"_PGC_umap_graph.tiff"), width = 700, height = 600)
  plot_cells(cds,
             color_cells_by = "orig_id",
             label_groups_by_cluster=FALSE,
             label_leaves=FALSE,
             label_branch_points=FALSE, cell_size = 1.5, 
             group_label_size = 7)
  dev.off()

  tiff(paste0(subgroup,"_PGC_umap_graph_time.tiff"), width = 700, height = 600)
  plot_cells(cds,
             color_cells_by = "time_point",
             label_groups_by_cluster=FALSE,
             label_leaves=FALSE,
             label_branch_points=FALSE, cell_size = 1.5, 
             group_label_size = 7)
  dev.off()

  tiff(paste0(subgroup,"_PGC_umap_graph_clusters.tiff"), width = 700, height = 600)
  plot_cells(cds,
             color_cells_by = "seu_cluster",
             label_groups_by_cluster=FALSE,
             label_leaves=FALSE,
             label_branch_points=FALSE, cell_size = 1.5, 
             group_label_size = 7)
  dev.off()

  cds <- order_cells(cds, root_cells = na.omit(rownames(cell.metadata)[cell.metadata$time_point == "E6.75"]))

  tiff(paste0(subgroup,"_PGC_umap_graph_pseudotime.tiff"), width = 700, height = 600)
  plot_cells(cds,
             color_cells_by = "pseudotime",
             label_cell_groups=FALSE,
             label_leaves=FALSE,
             label_branch_points=FALSE,
             cell_size = 1.5, 
             group_label_size = 7)
  dev.off()


  # Finding genes differentially expressed in pseudotime
  cds_test_res <- graph_test(cds, neighbor_graph="principal_graph", cores=4)
  deg_ids <- row.names(subset(cds_test_res, q_value < 0.05))
  
  # grouping genes in modules of co-expression
  gene_module_df <- find_gene_modules(cds[deg_ids,])
  gene_module_df_list[[subgroup]] <- gene_module_df
  
  cell_group_df <- tibble::tibble(cell=row.names(colData(cds)), 
                                  cell_group=colData(cds)$orig_id)
  agg_mat <- aggregate_gene_expression(cds, gene_module_df, cell_group_df)
  row.names(agg_mat) <- stringr::str_c("Module ", row.names(agg_mat))
  
  tiff(paste0(subgroup, "_heatmap_gene_modules.tiff"), width = 500, height = 600)
  pheatmap::pheatmap(agg_mat,
                     scale="column", clustering_method="ward.D2")
  dev.off()
  
  tiff(paste0(subgroup, "_gene_modules_umap.tiff"), width = 900, height = 700)
  plot_cells(cds,
             genes=gene_module_df,
             label_cell_groups=FALSE,
             show_trajectory_graph=FALSE,
             cell_size = 1.2, 
             group_label_size = 7)
  dev.off()


}


```


# Trying to use midhind which should not be clustering with MYC AT ALL

```{r cdng}
# MidHindbrain from Marioni
midhind <- readRDS("/marta_home/scRNAseq_Ranalysis/Seurat_v3/Embryos/published_data/v3/Marioni/full_marioni_integrated_midhind.rds")

for(i in 1:length(myc.list)){
  midhind.data <- GetAssayData(midhind, slot = "data")

  subgroup <- names(myc.list[i])
  myc.data <- GetAssayData(myc.list[[i]], slot = "data")
  
  # keep only intersecting genes
  genes.intersecting <- intersect(rownames(midhind.data), rownames(myc.data))

  midhind.data <- midhind.data[genes.intersecting,]
  myc.data <- myc.data[genes.intersecting,]

  merged.data <- cbind(midhind.data, myc.data)
  
  cell.metadata <- data.frame(orig_id = c(rep("MidHindbrain", ncol(midhind.data)), rep(subgroup, ncol(myc.data))), time_point = c(as.character(midhind$stage), rep(NA, ncol(myc.data))), seu_cluster = c(rep(NA, ncol(midhind.data)), as.character(myc.list[[i]]$new_clustering)), row.names = colnames(merged.data))

  gene.metadata <- data.frame(gene_short_name = rownames(merged.data), row.names = rownames(merged.data))

  # Make the CDS object
  cds <- new_cell_data_set(merged.data, cell_metadata = cell.metadata, gene_metadata = gene.metadata)

  # Preprocessing
  cds <- preprocess_cds(cds, num_dim = 100, norm_method = "none", method = "PCA")
  cds <- align_cds(cds, alignment_group = "orig_id")
                 
  # Dimensionality reduction and trajectories

  cds <- reduce_dimension(cds, reduction_method = "UMAP")
  tiff(paste0(subgroup, "_midhind_umap.tiff"), width = 700, height = 600)
  print(plot_cells(cds, label_groups_by_cluster=FALSE,  color_cells_by = "orig_id", cell_size = 1.5, 
           group_label_size = 7))
  dev.off()

  cds <- cluster_cells(cds)
  cds <- learn_graph(cds)

  tiff(paste0(subgroup,"_midhind_umap_graph.tiff"), width = 700, height = 600)
  print(plot_cells(cds,
             color_cells_by = "orig_id",
             label_groups_by_cluster=FALSE,
             label_leaves=FALSE,
             label_branch_points=FALSE, cell_size = 1.5, 
             group_label_size = 7))
  dev.off()

  tiff(paste0(subgroup,"_midhind_umap_graph_time.tiff"), width = 700, height = 600)
  print(plot_cells(cds,
             color_cells_by = "time_point",
             label_groups_by_cluster=FALSE,
             label_leaves=FALSE,
             label_branch_points=FALSE, cell_size = 1.5, 
             group_label_size = 7))
  dev.off()

  tiff(paste0(subgroup,"_midhind_umap_graph_clusters.tiff"), width = 700, height = 600)
  print(plot_cells(cds,
             color_cells_by = "seu_cluster",
             label_groups_by_cluster=FALSE,
             label_leaves=FALSE,
             label_branch_points=FALSE, cell_size = 1.5, 
             group_label_size = 7))
  dev.off()

  

}


```



# checking differences in marioni vs cao

```{r fnoa}
pgc.cao$orig.ident <- "cao"

plot1 <- VlnPlot(pgc.marioni, features = "nCount_RNA", group.by = "orig.ident")
plot2 <- VlnPlot(pgc.cao, features = "nCount_RNA", group.by = "orig.ident")
tiff("comparison_nCounts_PGCs.tiff")
CombinePlots(list(plot1, plot2), ncol = 2)
dev.off()
```



