---
title: "COO- BT16 DAC vs ctrl"
author: "Marta Interlandi"
date: "11/3/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r setup_environment, message=FALSE, warning=FALSE}

library(limma)
library(WriteXLS)

setwd("/marta_home/microarray/COO/")



```

## This are microarray data received from Pascal Johann (Heidelberg)


```{r gjoihg}
data <- read.table("./GEP_values_BT16_Decitabine.txt", dec = ",", stringsAsFactors = F)
data$V6 <- as.numeric(data$V6)

colnames(data) <- c("gene_name", "probe_ID", "control1", "control2", "control3", "DAC1", "DAC2", "DAC3", "Davon_gsk1", "Davon_gsk2", "Davon_gsk3", "Gsk126_1", "Gsk126_2", "Gsk126_3")
rownames(data) <- data$gene_name
data <- data[,-c(1,2)]

pheno <- data.frame(condition = c(rep("CTRL", 3), rep("DAC", 3), rep("dav", 3), rep("GSK",3)), row.names = colnames(data), stringsAsFactors = T)
norm.data <- normalizeBetweenArrays( log2(as.matrix(data)), method="quantile")
expr.data <- ExpressionSet(assayData = norm.data, phenoData = as(pheno, "AnnotatedDataFrame"))
boxplot(exprs(expr.data))


design <- model.matrix(~0 + factor(pheno$condition))
colnames(design)
colnames(design) <- c("ctrl", "DAC", "DAV", "GSK")
contrast.matrix <- makeContrasts(ctrl-DAC, levels = design)
fit <- lmFit(expr.data, design)
fit <- contrasts.fit(fit,contrast.matrix)
fit <- eBayes(fit)
head(fit$coefficients)
summary(decideTests(fit, adjust.method = "BH", p.value = 0.05))

#get full differential expression output table, sorted by p-value
limmaRes <- topTable(fit, coef= 1, adjust.method = "BH", n = Inf, sort.by = "P")
limmaRes$gene_name <- rownames(limmaRes)
```


```{r nrtuhi}

bt16_pascal <- read_excel("./BT16_DACvsctrl.xlsx")
min(bt16_pascal$DAC_vs_CTR)
bt16_pascal <- bt16_pascal$Gene_ID
unique(bt16_pascal) #96

bt16_wang <- read_excel("../../RNAseq/RT_cellLines_Wang_GSE71505/R/DE_BT16_noDoxvsDox.xlsx")
bt16_wang <- bt16_wang[bt16_wang$log2FC > 1.5, "gene"]
unique(bt16_wang$gene) #229 (1 NA)
intersect(bt16_pascal, bt16_wang$gene)

# 4 intersecting
```