---
title: "Mouse MRT MYC EC"
author: "Marta Interlandi"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r setup_environment, message=FALSE, warning=FALSE}
library(Seurat)
library(ggplot2)
library(sctransform)
library(RColorBrewer)
library(SeuratData)
library(readxl)
library(dplyr)
library(reshape2)
library(WriteXLS)
library(Hmisc)
library(pheatmap)

setwd("/marta_home/scRNAseq_Ranalysis/Seurat_v3/mouseATRT/v3/MRT_EC_MYC")

source("/marta_home/scRNAseq_Ranalysis/Seurat_v3/code/myaddModule_CellType.R")

load("/marta_home/scRNAseq_Ranalysis/Seurat_v3/mouseATRT/v3/MRT_EC_MYC/workspace.RData")
```

# Load datasets (UMI count matrix)
We load each dataset, filter out genes and merge them together before preprocessing

```{r load}
# Load the dataset

############## Extracranial

# Rosini1148 (MYC) -------------------------------------------------
data.matrix <- Read10X(data.dir = "/marta_home/10X/expMATRIX/_ATRT/mouse/EC_RT/Rosini1148/outs/filtered_gene_bc_matrices/mm10/")

dim(data.matrix)
head(colnames(data.matrix))
head(rownames(data.matrix))

colnames(data.matrix) <- paste("ros1148_cell", 1:length(colnames(data.matrix)), sep = "")
metadata <- data.frame(sample = rep("ros1148", ncol(data.matrix)), subgroup = rep("MYC", ncol(data.matrix)), localization = rep("EC", ncol(data.matrix)), row.names = colnames(data.matrix))


# Create a Seurat object
# Keep all cells with at least 50 detected genes
ros1148 <- CreateSeuratObject(counts = data.matrix, min.cells = 3, min.features = 50, meta.data = metadata)
ros1148

write.table(ros1148@assays$RNA@counts, file = "/marta_home/GEO_COO/single_cell/Rosini1148_rawUMI.txt", sep = "\t", quote = F)

# Soxertini568 (MYC) -------------------------------------------------
data.matrix <- Read10X(data.dir = "/marta_home/10X/expMATRIX/_ATRT/mouse/EC_RT/Soxertini568_maxilar/outs/filtered_gene_bc_matrices/mm10/")

dim(data.matrix)
head(colnames(data.matrix))
head(rownames(data.matrix))

colnames(data.matrix) <- paste("sox568_cell", 1:length(colnames(data.matrix)), sep = "")
metadata <- data.frame(sample = rep("sox568", ncol(data.matrix)), subgroup = rep("MYC", ncol(data.matrix)), localization = rep("EC", ncol(data.matrix)), row.names = colnames(data.matrix))


# Create a Seurat object
# Keep all cells with at least 50 detected genes
sox568 <- CreateSeuratObject(counts = data.matrix, min.cells = 3, min.features = 50, meta.data = metadata)
sox568

write.table(sox568@assays$RNA@counts, file = "/marta_home/GEO_COO/single_cell/Soxertini568_rawUMI.txt", sep = "\t", quote = F)


# Soxertini593 (MYC) -------------------------------------------------
data.matrix <- Read10X(data.dir = "/marta_home/10X/expMATRIX/_ATRT/mouse/EC_RT/Soxertini593/outs/filtered_gene_bc_matrices/mm10/")

dim(data.matrix)
head(colnames(data.matrix))
head(rownames(data.matrix))

colnames(data.matrix) <- paste("sox593_cell", 1:length(colnames(data.matrix)), sep = "")
metadata <- data.frame(sample = rep("sox593", ncol(data.matrix)), subgroup = rep("MYC", ncol(data.matrix)), localization = rep("EC", ncol(data.matrix)), row.names = colnames(data.matrix))


# Create a Seurat object
# Keep all cells with at least 50 detected genes
sox593 <- CreateSeuratObject(counts = data.matrix, min.cells = 3, min.features = 50, meta.data = metadata)
sox593

write.table(sox593@assays$RNA@counts, file = "/marta_home/GEO_COO/single_cell/Soxertini593_rawUMI.txt", sep = "\t", quote = F)

```


# Quality control and filtering 

```{r vnifo}
tumor.list <- list(ros1148 = ros1148, sox568 = sox568, sox593 = sox593)

for (i in 1:length(tumor.list)) {
    tumor.list[[i]][["percent.mt"]] <- PercentageFeatureSet(tumor.list[[i]], pattern = "^mt-")
    plot1 <- FeatureScatter(tumor.list[[i]], feature1 = "nCount_RNA", feature2 = "percent.mt", cols = "black")
    plot2 <- FeatureScatter(tumor.list[[i]], feature1 = "nCount_RNA", feature2 = "nFeature_RNA", cols = "black")
    tiff(paste0("./preprocessing/", names(tumor.list[i]), ".tiff"), width = 500, height = 400)
    print(CombinePlots(plots = list(plot1, plot2), legend = 'none'))
    dev.off()
}

for (i in 1:length(tumor.list)) {
  tumor.list[[i]] <- subset(tumor.list[[i]], subset = nFeature_RNA > 200 & percent.mt < 25)
}



```

# Normalize the data with log normalization


```{r nvffr}
for (i in 1:length(tumor.list)) {
  tumor.list[[i]] <- NormalizeData(tumor.list[[i]])
  tumor.list[[i]] <- FindVariableFeatures(tumor.list[[i]], selection.method = "vst", 
        nfeatures = 2000)
}


```

# Integration 

```{r vcmnvf}
atrt.anchors <- FindIntegrationAnchors(object.list = tumor.list, dims = 1:30, k.filter = 150)

features.intersection <- c(Reduce(intersect, lapply(atrt.anchors@object.list, rownames)))
atrt.integrated <- IntegrateData(anchorset = atrt.anchors, features.to.integrate = features.intersection)


```

# Plot UMAP

```{r cdi}

atrt.integrated <- ScaleData(atrt.integrated)
atrt.integrated <- RunPCA(atrt.integrated)
atrt.integrated <- RunUMAP(atrt.integrated, dims = 1:30)
plots <- DimPlot(atrt.integrated, group.by = c("sample", "subgroup"), combine = FALSE, pt.size = 1.5)
plots <- lapply(X = plots, FUN = function(x) x + theme(legend.position = "top") + guides(color = guide_legend(nrow = 3, 
    byrow = TRUE, override.aes = list(size = 3))))
tiff("atrt_MYCec_integrated_sample_sub.tiff", width = 1200, height = 800)
print(CombinePlots(plots))
dev.off()

tiff("atrt_MYCec_integrated_sample.tiff", width = 1200, height = 500)
DimPlot(atrt.integrated, split.by = "sample", group.by = "orig.ident", ncol = 5, pt.size = 0.8)
dev.off()

tiff("atrt_MYCec_integrated.tiff", width = 900, height = 700)
DimPlot(atrt.integrated, group.by = "orig.ident", pt.size = 0.8)
dev.off()


tiff("atrt_MYCec_nUMI.tiff", width = 900, height = 700)
FeaturePlot(object = atrt.integrated,  features = "nCount_RNA", pt.size = 0.8)
dev.off()

tiff("atrt_MYCec_nGene.tiff", width = 900, height = 700)
FeaturePlot(object = atrt.integrated, features = "nFeature_RNA", pt.size = 0.8)
dev.off()


DefaultAssay(atrt.integrated) <- "integrated"

atrt.integrated <- FindNeighbors(atrt.integrated, reduction = "pca", dims = 1:30)
atrt.integrated <- FindClusters(atrt.integrated, resolution = 0.5)
tiff("atrt_MYCec_clusters.tiff", width = 900, height = 700)
DimPlot(atrt.integrated, pt.size = 0.8, group.by = "integrated_snn_res.0.5", label = T, label.size = 6)
dev.off()

# Create a table with number of cells by sample / cluster
df <- data.frame(sample = as.character(atrt.integrated$orig.ident), 
                 cluster = as.character(atrt.integrated$integrated_snn_res.0.5), 
                  stringsAsFactors = F)
Summary <- df %>%
  group_by(cluster, sample) %>%
  dplyr::summarize(n())
final <- recast(Summary, cluster~sample)
WriteXLS(final, ExcelFileName = "cellNumber_sampleBYclusters.xlsx", row.names = F)
  

```



# Plot markers


```{r cndiofs}
DefaultAssay(atrt.integrated) <- "RNA"


tiff("atrt_MYCec_selected_markers.tiff", width = 1000, height = 900)
FeaturePlot(atrt.integrated, c("Smarcb1", "Bmp4", "Mycn", "Epcam", "Cd44", "Cd68", "P2ry12", "Ms4a7", "Col1a2"))
dev.off()

VlnPlot(atrt.integrated, features = "Smarcb1")
smarcbp.cells <- WhichCells(atrt.integrated, expression = Smarcb1 > 0.2)
tiff("atrt_MYCec_Smarcb1_pos_cells.tiff", width = 900, height = 700)
DimPlot(object = atrt.integrated, reduction.use = "tsne", pt.size = 0.8, cells.highlight = smarcbp.cells, cols.use = "gray", cols.highlight = "darkred", order = smarcbp.cells) + ggtitle("SMARCB1 positive cells")
dev.off()

tiff("MYC_EC_Dnd1.tiff", width = 900, height = 700)
FeaturePlot(atrt.integrated, features = "Dnd1", pt.size = .8)
dev.off()

```


# Gene expression programs

```{r annot}
# markers final
list.mark <- read_excel("/marta_home/scRNAseq_Ranalysis/markers/markers_ATRTfinal2.xlsx", col_names = T)
list.mark <- list.mark[, -c(2,3,4)]

genes.list <- list("Tumor_MYC" = list.mark$`Tumor MYC IC/EC/spinal`[!is.na(list.mark$`Tumor MYC IC/EC/spinal`)], 
                   "Microglia-derived_TAMs" = list.mark$`Microglia-derived TAMs`[!is.na(list.mark$`Microglia-derived TAMs`)],
                   "Blood-derived_TAMs" = list.mark$`Blood-derived TAMs`[!is.na(list.mark$`Blood-derived TAMs`)],
                   "Dendritic_cells" = list.mark$`Dendritic cells`[!is.na(list.mark$`Dendritic cells`)],
                   "Granulocytes_Mastcells" = list.mark$`Granulocytes/Mast cells`[!is.na(list.mark$`Granulocytes/Mast cells`)],
                   "NKcells" = list.mark$`NK cells`[!is.na(list.mark$`NK cells`)],
                   "Bcells" = list.mark$`B cells`[!is.na(list.mark$`B cells`)],
                   "Tcells" = list.mark$`T cells`[!is.na(list.mark$`T cells`)],
                   "Oligo" = list.mark$Oligo[!is.na(list.mark$Oligo)])


atrt.integrated <- myaddModule_CellType(atrt.integrated, genes.list)

# Create vector of colors
my_color_palette <- RColorBrewer::brewer.pal(length(unique(atrt.integrated$CellType)), "Paired")
my_color_palette[10] <- "slategrey"

tiff("atrt_MYCec_annotation.tiff", width = 1000, height = 700)
DimPlot(object = atrt.integrated , pt.size = 1, cols = my_color_palette, group.by = "CellType")  +  theme(legend.text=element_text(size=13))
dev.off()



```

## Find markers

```{r cd}
DefaultAssay(atrt.integrated) <- "RNA"

for (i in levels(Idents(atrt.integrated))){
  markers <- FindMarkers(atrt.integrated, ident.1 = i, test.use = "MAST")
  markers <- markers[markers$p_val_adj < 0.05, ]
  markers <- markers[!is.na(markers$p_val_adj), ]
  WriteXLS(markers, ExcelFileName = paste0("MYC_EC_markers_", i, ".xlsx"), row.names = T)
}


```



### Plot tumor markers from Moni's table

```{r m}

mark.tab <- read_excel("/marta_home/scRNAseq_Ranalysis/Seurat_v3/markers/subgroup specific marker.xlsx", sheet = "Tabelle1", skip = 1, col_names = T)

group.name <- colnames(mark.tab)

for(col in group.name){
  markers <- as.character(unlist(mark.tab[, col]))
  markers <- markers[!is.na(markers)]
  markers <- markers[markers %in% rownames(atrt.integrated)]
  col <- gsub(" ", "_", col)
  col <- gsub("/", "_", col)
  for(mark in markers){
  tiff(paste0("./tumor_markers/tabelle1/", col, "_", mark, "_feat.tiff"), width = 900, height = 700)
  print(FeaturePlot(atrt.integrated, features = mark, pt.size = 1, cols = c("gray", "blue")))
  dev.off()
  }
}



mark.tab <- read_excel("/marta_home/scRNAseq_Ranalysis/Seurat_v3/markers/subgroup specific marker.xlsx", sheet = "Tab2_short", col_names = T)

group.name <- colnames(mark.tab)

for(col in group.name){
  markers <- as.character(unlist(mark.tab[, col]))
  markers <- markers[!is.na(markers)]
  markers <- markers[markers %in% rownames(atrt.integrated)]
  col <- gsub(" ", "_", col)
  col <- gsub("/", "_", col)
  for(mark in markers){
  tiff(paste0("./tumor_markers/tabelle2/", col, "_", mark, "_feat.tiff"), width = 900, height = 700)
  print(FeaturePlot(atrt.integrated, features = mark, pt.size = 1, cols = c("gray", "blue")))
  dev.off()
  }
}


mark.tab <- read_excel("/marta_home/scRNAseq_Ranalysis/Seurat_v3/markers/subgroup specific marker.xlsx", sheet = "Tabelle3", col_names = T)

group.name <- colnames(mark.tab)

for(col in group.name){
  markers <- as.character(unlist(mark.tab[, col]))
  markers <- markers[!is.na(markers)]
  markers <- capitalize(tolower(markers)) 
  markers <- markers[markers %in% rownames(atrt.integrated)]
  col <- gsub(" ", "", col)
  col <- gsub(",", "", col)
  for(mark in markers){
  tiff(paste0("./tumor_markers/tabelle3/", col, "_", mark, "_feat.tiff"), width = 900, height = 700)
  print(FeaturePlot(atrt.integrated, features = mark, pt.size = 1, cols = c("gray", "blue")))
  dev.off()
  }
}


```

## Plot Immune markers from annotation table

```{r cndufig}

list.mark <- read_excel("/marta_home/scRNAseq_Ranalysis/markers/markers_ATRTfinal2.xlsx", col_names = T)
list.mark <- list.mark[, -c(1:4)]
group.name <- colnames(list.mark)

for(col in group.name){
  markers <- as.character(unlist(list.mark[, col]))
  markers <- markers[!is.na(markers)]
  markers <- capitalize(tolower(markers)) 
  markers <- markers[markers %in% rownames(atrt.integrated)]
  col <- gsub(" ", "", col)
  col <- gsub("/", "_", col)
  for(mark in markers){
  tiff(paste0("./immune_markers/", col, "_", mark, ".tiff"), width = 900, height = 700)
  print(FeaturePlot(atrt.integrated, features = mark, pt.size = 1, cols = c("gray", "blue"), order = T))
  dev.off()
  }
}


```


## Subcluster cluster 8

```{r cmpgb}
clust8 <- subset(atrt.integrated, subset = seurat_clusters == "8")
DefaultAssay(clust8) <- "integrated"

clust8 <- RunPCA(clust8)
clust8 <- FindNeighbors(clust8, reduction = "pca", dims = 1:20)
clust8 <- FindClusters(clust8, resolution = 0.2)
tiff("atrt_MYCec_clust8_subclustered.tiff", width = 900, height = 700)
DimPlot(clust8, pt.size = 0.8, group.by = "integrated_snn_res.0.2", label = T, label.size = 6)
dev.off()


cells.tum.clust8 <- WhichCells(clust8, idents = 1)

# Rename identities in full dataset
Idents(atrt.integrated, cells = cells.tum.clust8) <- 18

atrt.integrated$new_clustering <- factor(Idents(atrt.integrated), levels = 0:18)

tiff("atrt_MYCec_NEWclusters.tiff", width = 900, height = 700)
DimPlot(atrt.integrated, pt.size = 0.8, group.by = "new_clustering", label = T, label.size = 6)
dev.off()

tiff("atrt_MYCec_NEWclusters_noNum.tiff", width = 900, height = 700)
DimPlot(atrt.integrated, pt.size = 0.8, group.by = "new_clustering", label = F, label.size = 6)
dev.off()


## Recompute DE lists for new clusters
DefaultAssay(atrt.integrated) <- "RNA"

# clust 8
markers <- FindMarkers(atrt.integrated, ident.1 = 8, test.use = "MAST")
markers <- markers[markers$p_val_adj < 0.05, ]
markers <- markers[!is.na(markers$p_val_adj), ]
WriteXLS(markers, ExcelFileName = paste0("MYC_EC_markers_", 8, ".xlsx"), row.names = T)

# clust 18
markers <- FindMarkers(atrt.integrated, ident.1 = 18, test.use = "MAST")
markers <- markers[markers$p_val_adj < 0.05, ]
markers <- markers[!is.na(markers$p_val_adj), ]
WriteXLS(markers, ExcelFileName = paste0("MYC_EC_markers_", 18, ".xlsx"), row.names = T)

```
# Create a table with number of cells by cluster / cell type

```{r vnfio}
df <- data.frame(cluster = as.character(atrt.integrated$new_clustering), 
                 celltype = as.character(atrt.integrated$CellType), 
                  stringsAsFactors = F)
Summary <- df %>%
  group_by(cluster, celltype) %>%
  dplyr::summarize(n())
final <- recast(Summary, celltype~cluster)
WriteXLS(final, ExcelFileName = "MYC_EC_cellNumber_celltypeBYcluster.xlsx", row.names = F)
  
``` 

## Save tumor clusters for COO

```{r dmisobf}
tum.cells <- colnames(atrt.integrated)[atrt.integrated$new_clustering %in% c(14,16,5,12,18)]
tum.clusts <- atrt.integrated[, tum.cells]
dim(GetAssay(tum.clusts, "RNA"))


saveRDS(tum.clusts, file = "myc_ec_tumor_cells.rds")


```
## DE tumor clusters vs NON-tumor clusters

```{r cndiso}
DefaultAssay(atrt.integrated) <- "RNA"
markers <- FindMarkers(atrt.integrated, ident.1 = c(14,16,5,12,18), test.use = "MAST")
markers <- markers[markers$p_val_adj < 0.05, ]
markers <- markers[!is.na(markers$p_val_adj), ]
WriteXLS(markers, ExcelFileName = "DE_MYC_EC_tumorVSothers.xlsx", row.names = T)

```


## Save entire object for plotting markers

```{r dmisobf}

saveRDS(atrt.integrated, file = "myc_ec_seuobj.rds")


```


# Moni heatmap tumor cluster

```{r vobpdbh}
gene.tab <- read_excel("/marta_home/scRNAseq_Ranalysis/Seurat_v3/mouseATRT/markers/markers_heatmap_MYCec.xlsx")

gene.categ <- colnames(gene.tab)

genes.all <- c()
for(cat in gene.categ){
  genes <- capitalize(tolower(as.character(unlist(gene.tab[, cat]))))
  genes <- genes[genes %in% rownames(tum.clusts)]
  genes.all <- c(genes.all, genes)
}

# check for duplicated genes and delete them
dup.genes <- genes.all[duplicated(genes.all)]

genes.all.nodup <- c()
gaps <- c()

for(cat in gene.categ){
  genes <- capitalize(tolower(as.character(unlist(gene.tab[, cat]))))
  genes <- genes[genes %in% rownames(tum.clusts)]
  genes <- genes[!(genes %in% dup.genes)]
  genes.all.nodup <- c(genes.all.nodup, genes)
  gaps <- c(gaps, length(genes))
}
sum(duplicated(genes.all.nodup))

WriteXLS(x = as.data.frame(genes.all.nodup), ExcelFileName = "MYC_EC_heat_genesNOdup.xlsx")

tum_clust_mat <- GetAssayData(tum.clusts, slot = "data", assay = "RNA")
dim(tum_clust_mat)
tum_clust_mat <- tum_clust_mat[genes.all.nodup,]
dim(tum_clust_mat)

# order cells by cluster
clusts <- sort(tum.clusts$new_clustering)
head(clusts)
tum_clust_mat <- tum_clust_mat[, names(clusts)]

annotation_col <- data.frame(cluster = clusts, row.names = names(clusts))
annotation_row <- data.frame(gene_category = rep(gene.categ, times = gaps), row.names = genes.all.nodup)

pheatmap(tum_clust_mat, scale = "none", cluster_rows = F, cluster_cols = F,  annotation_row = annotation_row, annotation_col = annotation_col,annotation_names_row = F, gaps_row = cumsum(gaps), show_rownames = F, show_colnames = F, filename = "MYC_EC_tumHeatmap.tiff", width = 10, height = 6)


## aggregate cluster by computing average

mat <- as.data.frame(t(as.matrix(tum_clust_mat)))

mat$cluster <- clusts

mean.mat <- aggregate(. ~ cluster, mat, mean)
rownames(mean.mat) <- mean.mat$cluster

annotation_col <- data.frame(cluster= mean.mat$cluster, row.names = rownames(mean.mat))

mean.mat <- mean.mat[,-1]
mean.mat <- t(as.matrix(mean.mat))

pheatmap(mean.mat, scale = "row", cluster_rows = F, cluster_cols = F,  annotation_row = annotation_row, annotation_col = annotation_col, annotation_names_row = F, gaps_row = cumsum(gaps), show_rownames = F, show_colnames = F, filename = "MYC_EC_tumHeatmap_mean.tiff", width = 10, height = 6)



```


## Plot PGC markers

```{r bkgopsh}

pgc.mark <- c("Pou5f1", "Tfap2c", "Prdm1")

for(mark in pgc.mark){
  tiff(paste0("./PGCmarkers/", mark, "_feat.tiff"), width = 900, height = 700)
  print(FeaturePlot(atrt.integrated, features = mark, pt.size = 1, cols = c("gray", "blue"), order = T))
  dev.off()
}




```




## DE analysis only tumors

```{r cnifoga}
DefaultAssay(tum.clusts) <- "RNA"

for (i in levels(Idents(tum.clusts))){
  markers <- FindMarkers(tum.clusts, ident.1 = i, test.use = "MAST")
  markers <- markers[markers$p_val_adj < 0.05, ]
  markers <- markers[!is.na(markers$p_val_adj), ]
  WriteXLS(markers, ExcelFileName = paste0("MYC_EC_markers_TUMonly_", i, ".xlsx"), row.names = T)
}


```

## Plot predictor genes

```{r cnduig}
plot_predictors <- function(pred.df, celltype, data){
  pred.genes.pos <- as.character(unlist(pred.df[pred.df$coefficient >0, "predictors"]))
  pred.genes.neg <- as.character(unlist(pred.df[pred.df$coefficient <0, "predictors"]))
  
  pred.genes.pos <- pred.genes.pos[pred.genes.pos %in% rownames(data)]
  pred.genes.neg <- pred.genes.neg[pred.genes.neg %in% rownames(data)]
  
  for(mark in pred.genes.pos){
    tiff(paste0("./predictors/", celltype, "/positive/", mark, "_feat.tiff"), width = 900, height = 700)
    print(FeaturePlot(data, features = mark, pt.size = 1, cols = c("gray", "blue"), order = T))
    dev.off()
  }
  
  for(mark in pred.genes.neg){
    tiff(paste0("./predictors/", celltype,"/negative/", mark, "_feat.tiff"), width = 900, height = 700)
    print(FeaturePlot(data, features = mark, pt.size = 1, cols = c("gray", "blue"), order = T))
    dev.off()
  }
}

## PGCs
pgc.pred.genes <- read_xlsx("/marta_home/scRNAseq_Ranalysis/Seurat_v3/COO-logisticReg/binomialLR/PGC_predictors.xlsx")

plot_predictors(pgc.pred.genes, "PGC", atrt.integrated)

## Midhindbrain
midhind.pred.genes <- read_xlsx("/marta_home/scRNAseq_Ranalysis/Seurat_v3/COO-logisticReg/binomialLR/MidHindbrain_predictors.xlsx")
plot_predictors(midhind.pred.genes, "midhind", atrt.integrated)
```


#### Cell cycle scoring

```{r fneiog}
cell.cycle.genes <- read_excel("/marta_home/scRNAseq_Ranalysis/markers/cell_cycle_markers_Tirosh.xlsx", skip = 1)
s.genes <- cell.cycle.genes$`G1/S`
g2m.genes <- cell.cycle.genes$`G2/M`

s.genes <- s.genes[!is.na(s.genes)]
g2m.genes <- g2m.genes[!is.na(g2m.genes)]


s.genes <- capitalize(tolower(s.genes))
g2m.genes <- capitalize(tolower(g2m.genes))

s.genes <- s.genes[s.genes %in% rownames(atrt.integrated)]
g2m.genes <- g2m.genes[g2m.genes %in% rownames(atrt.integrated)]

atrt.integrated <- CellCycleScoring(atrt.integrated, s.features = s.genes, g2m.features = g2m.genes)

tiff("atrt_MYC_EC_CCphase.tiff", width = 900, height = 700)
DimPlot(atrt.integrated, pt.size = 0.8, group.by = "Phase", cols = c("#1f77b4", "#ff7f0e", "#279e68"))
dev.off()


# Rename idents in full ds
cc_cells <- colnames(atrt.integrated)[atrt.integrated$new_clustering == "18"]

full_CC_clusts <- atrt.integrated$new_clustering
full_CC_clusts[cc_cells] <- NA

atrt.integrated$CC_clusts <- full_CC_clusts

tiff("atrt_MYC_EC_CC_clusters_all.tiff", width = 900, height = 700)
DimPlot(atrt.integrated, pt.size = 0.8, group.by = "CC_clusts", label = F)
dev.off()
```
# Moni heatmap tumor cluster CC clust CC functions

```{r vobpdbh}
gene.tab <- read_excel("/marta_home/scRNAseq_Ranalysis/Seurat_v3/mouseATRT/v3/genes_CC_MYC_heatmap_updated.xlsx", sheet = "EC")
gene.categ <- colnames(gene.tab)

genes.all <- c()
gaps <- c()
for(cat in gene.categ){
  genes <- capitalize(tolower(as.character(unlist(gene.tab[, cat]))))
  genes <- genes[genes %in% rownames(tum.clusts)]
  genes.all <- c(genes.all, genes)
  gaps <- c(gaps, length(genes))
}

tum_clust_mat <- GetAssayData(tum.clusts, slot = "data", assay = "RNA")
dim(tum_clust_mat)
tum_clust_mat <- tum_clust_mat[genes.all,]
dim(tum_clust_mat)

# order cells by cluster
clusts <- sort(tum.clusts$new_clustering)
head(clusts)
tum_clust_mat <- tum_clust_mat[, names(clusts)]

annotation_col <- data.frame(cluster = clusts, row.names = names(clusts))
annotation_row <- data.frame(gene_category = rep(gene.categ, times = gaps), row.names = genes.all)

pheatmap(tum_clust_mat, scale = "none", color= colorRampPalette(brewer.pal(n = 9, name = "Greys"))(50), cluster_rows = F, cluster_cols = F,  annotation_row = annotation_row, annotation_col = annotation_col, annotation_names_row = F, gaps_row = cumsum(gaps), show_rownames = T, show_colnames = F, filename = "MYC_EC_tumHeatmap_CCall.tiff", width = 10, height = 6)

## aggregate cluster by computing average

mat <- as.data.frame(t(as.matrix(tum_clust_mat)))

mat$cluster <- clusts

mean.mat <- aggregate(. ~ cluster, mat, mean)
rownames(mean.mat) <- mean.mat$cluster

annotation_col <- data.frame(cluster= mean.mat$cluster, row.names = rownames(mean.mat))

mean.mat <- mean.mat[,-1]
mean.mat <- t(as.matrix(mean.mat))

pheatmap(mean.mat, scale = "none", color= colorRampPalette(brewer.pal(n = 9, name = "Greys"))(50), cluster_rows = F, cluster_cols = F,  annotation_row = annotation_row, annotation_col = annotation_col, annotation_names_row = F, gaps_row = cumsum(gaps), show_rownames = T, show_colnames = F, filename = "MYC_EC_tumHeatmap_CCall_mean.tiff", width = 10, height = 6)


##---- plotting each column separately
tum_clust_mat <- GetAssayData(tum.clusts, slot = "data", assay = "RNA")
dim(tum_clust_mat)
# order cells by cluster
clusts <- sort(tum.clusts$new_clustering)
head(clusts)
tum_clust_mat <- tum_clust_mat[, names(clusts)]
annotation_col <- data.frame(cluster = clusts, row.names = names(clusts))


for(cat in gene.categ){
  genes <- capitalize(tolower(as.character(unlist(gene.tab[, cat]))))
  genes <- genes[genes %in% rownames(tum.clusts)]
  
  tum_clust_sub <- tum_clust_mat[genes,]
  
  pheatmap(tum_clust_sub, color= colorRampPalette(brewer.pal(n = 9, name = "Greys"))(50), scale = "none", cluster_rows = F, cluster_cols = F,  annotation_col = annotation_col, show_rownames = T, show_colnames = F, filename = paste0("MYC_EC_tumHeatmap_CC_", cat, ".tiff"), width = 10, height = 6)
}




```
# Dotplot epi markers tum clusters


```{r cndsif}
epi.mark <- c("Ezh2", "Dnmt3a", "Dnmt1", "Tet1", "Kdm6a")

tiff("MYC_EC_dotplot_EPIgenes_tumors4Fig.tiff", width = 300, height = 400)
DotPlot(tum.clusts, assay = "RNA", features = epi.mark, group.by = "new_clustering", cols = c("aquamarine", "darkblue")) + theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) + scale_x_discrete(limits = levels(factor(epi.mark)))
dev.off()
```


# Dotplot stromal markers tumor vs tme


```{r cndsif}

stromal.mark <- read_excel("../marker list stromal markers.xlsx")
stromal.mark <- unlist(stromal.mark)
stromal.mark <- capitalize(tolower(stromal.mark))
stromal.mark <- stromal.mark[stromal.mark %in% rownames(atrt.integrated)]

# Add Smarcb1 
stromal.mark <- c("Smarcb1", stromal.mark)




pheno <- as.character(atrt.integrated$seurat_clusters)
names(pheno) <- colnames(atrt.integrated)
tum.cells <- colnames(atrt.integrated)[atrt.integrated$new_clustering %in% c(14,16,5,12,18)]
pheno[] <- "TME"
pheno[tum.cells] <- "TUM"
table(pheno)
atrt.integrated$pheno <- as.factor(pheno)

pdf("MYC_EC_dotplot_stromal_TUMvsTME.pdf")
DotPlot(atrt.integrated, features = stromal.mark, group.by = "pheno", dot.scale = 10) + theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 15)) 
dev.off()


```

